search/materialize.go

```diff
@@ -164,6 +180,56 @@ func (m *Materializer) Materialize(ctx context.Context, rgi int, mint, maxt int6
 	return results, err
 }
 
+func (m *Materializer) filterSeries(ctx context.Context, hints *prom_storage.SelectHints, sLbls [][]labels.Label, rr []RowRange) ([]prom_storage.ChunkSeries, []RowRange) {
+	results := make([]prom_storage.ChunkSeries, 0, len(sLbls))
+	labelsFilter, ok := m.materializedLabelsFilterCallback(ctx, hints)
+	if !ok {
+		for _, s := range sLbls {
+			results = append(results, &concreteChunksSeries{
+				lbls: labels.New(s...),
+			})
+		}
+		return results, rr
+	}
+
+	defer labelsFilter.Close()
+
+	// Convert matching rows back into row ranges
+	currentRange := RowRange{}
+	inRange := false
+	filteredRR := make([]RowRange, 0, len(rr))
+	for i, s := range sLbls {
+		lbls := labels.New(s...)
+		if labelsFilter.Filter(lbls) {
+			results = append(results, &concreteChunksSeries{
+				lbls: lbls,
+			})
+			if !inRange {
+				// Start a new range
+				currentRange.From = int64(i)
```

Comment:
`i` goes from `0` to `len(sLbls)`, those are not necessarily row numbers. I would expect this loop to use the original `rr` somewhere.

Or am I missing something? 🤔